app/components/blacklight/gallery/openseadragon_embed_component.rb

```diff
@@ -3,15 +3,19 @@
 module Blacklight
   module Gallery
     class OpenseadragonEmbedComponent < Blacklight::Component
-      attr_reader :document, :presenter, :classes
+      attr_reader :document, :classes
 
-      def initialize(document:, presenter:, view_config: nil, classes: [], **kwargs)
+      def initialize(document:, view_config: nil, classes: [], **kwargs)
         super
 
         @document = document
-        @presenter = presenter
         @view_config = view_config
         @classes = classes
+        @id_prefix = generate_id_prefix
+      end
+
+      def presenter
+        @presenter ||= @view_context.document_presenter(@document)
```

Comment:
Why are we not continuing to pass in the presenter?  Doesn't this cause an unnecessary object allocation?